alg/llrasterize.cpp

```diff
@@ -498,6 +503,11 @@ GDALdllImageLineAllTouched( int nRasterXSize, int nRasterYSize,
             // Special case for horizontal lines.
             if( floor(dfY) == floor(dfYEnd) || fabs(dfY - dfYEnd) < .01 )
             {
+                double dYOffset = std::abs(std::fmod(dfY, 1.));
+                double dYEndOffset = std::abs(std::fmod(dfYEnd, 1.));
+                if ( bIntersectOnly && abs(dYOffset - .5) > .49 &&  abs(dYEndOffset - .5) > .49)
+                    continue;
```

Comment:
to save some computations when bIntersectOnly == false
```suggestion
                if ( bIntersectOnly )
                {
                    double dYOffset = std::abs(std::fmod(dfY, 1.));
                    double dYEndOffset = std::abs(std::fmod(dfYEnd, 1.));
                    if( std::abs(dYOffset - .5) > .49 &&  std::abs(dYEndOffset - .5) > .49)
                        continue;
                }
```

but couldn't that be just rewritten as
```suggestion
                if ( bIntersectOnly )
                {
                    if( std::abs(dfY - std::round(dfY)) < 0.01 && std::abs(dfYEnd - std::round(dfYEnd)) < 0.01 )
                        continue;
                }
```